File: PiDevMobile/src/com/mycompany/entites/event.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycompany.entites;

import java.util.Date;

/**
 *
 * @author bhk
 */
public class event {
     private int id_event;
    private int id_coach;
    private String lieu;
    private int nb_place   ;
   private String date_event ;
   private String description;
     private String Categorie ;
    
    public event(int id_event, int id_coach,String lieu,int nb_place,String date_event,String description,String Categorie) {
        this.id_event = id_event;
        this.id_coach = id_coach;
         this.lieu = lieu;

        this.nb_place = nb_place;
     this.date_event = date_event;
          this.description = description;

        this.Categorie = Categorie;


    }
    public event(){}
public event( String lieu,int nb_place,String date_event,String description,String Categorie) {
       
        this.lieu = lieu;
        this.nb_place = nb_place;
        this.date_event = date_event;
                this.description = description;

        this. Categorie =  Categorie;


    }
public event( String lieu,int nb_place,String date_event,String description) {
       
        this.lieu = lieu;
        this.nb_place = nb_place;
        this.date_event = date_event;
                             this.description = description;



    }

  


   

    public String getDescription() {
        return description;
    }

    /* public event(int aInt, String string) {
    throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
    }*/
    public void setDescription(String description) {
        this.description = description;
    }

    public int getId_event() {
        return id_event;
    }

    public void setId_event(int id_event) {
        this.id_event = id_event;
    }

    public int getId_coach() {
        return id_coach;
    }

    public void setId_coach(int id_coach) {
        this.id_coach = id_coach;
    }

   

    public String getLieu() {
        return lieu;
    }

    public void setLieu(String lieu) {
        this.lieu = lieu;
    }

    public int getNb_place() {
        return nb_place;
    }

    public void setNb_place(int nb_place) {
        this.nb_place = nb_place;
    }

   public String getDate_event() {
       return date_event;
    }

    public void setDate_event(String date_event) {
       this.date_event = date_event;
    }

    public String getCategorie() {
        return  Categorie;
    }

    public void setCategorie(String  Categorie) {
        this. Categorie =  Categorie;
    }

    @Override
    public String toString() {
        return "event{" + "id_event=" + id_event + ", id_coach=" + id_coach + ", lieu=" + lieu + ", nb_place=" + nb_place + ", date_event=" + date_event + ",  description=" + description + ", Categorie=" +  Categorie + '}';
    }

  
  
}
//, date_event=" + date_event + "

File: PiDevWebFinal/src/Entity/Event.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use phpDocumentor\Reflection\File;
use PhpParser\Node\Scalar\String_;
use Symfony\Component\Validator\Constraints as Assert;
use Vich\UploaderBundle\Mapping\Annotation as Vich;
use Symfony\Component\Validator\Constraints\DateTime;
use Symfony\Component\Serializer\Annotation\Groups;



/**
 * Event
 *
 * @ORM\Table(name="event", indexes={@ORM\Index(name="fk_idcatE", columns={"category"})})
 * @ORM\Entity
 * @Vich\Uploadable()
 */
class Event
{
    /**
     * @var int
     *
     * @ORM\Column(name="id_event", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     * @Groups ("post:read")
     *
     */
    private $idEvent;

    /**
     * @return File
     */
    public function getImageFile()
    {
        return $this->imageFile;
    }

    /**
     * @param File $imageFile
     */
    public function setImageFile($imageFile=null): void
    {
        $this->imageFile = $imageFile;

        if ($imageFile) {
            $this->updatedAt = new DateTime('now');
        }
    }




   /**
     * @var File
     * @Vich\UploadableField(mapping="images",fileNameProperty="description")
     *@Assert\NotBlank(message="Image obligatoire")
     * @Groups ("post:read")
     */

    private $imageFile;
    /**
     * @var String
     * @ORM\Column(type="string")
     * @Groups ("post:read")
     */
    private $description;


    /**
     * @var int|null
     * @ORM\Column(name="id_coach", type="integer", nullable=true)
     * @Groups ("post:read")
     */
    private $idCoach;

    /**
     * @var string|null
     *
     * @ORM\Column(name="lieu", type="string", length=255, nullable=true)
     * @Groups ("post:read")
     */
    private $lieu;

    /**
     * @var int|null
     *
     * @ORM\Column(name="nb_place", type="integer", nullable=true)
     * @Assert\NotBlank(message="nombre de place obligatoire")
     * @Groups ("post:read")
     */
    private $nbPlace;

    /**
     * @var \DateTime|null
     *
     * @ORM\Column(name="date_event", type="date", nullable=true)
     * @Assert\NotBlank(message="Date obligatoire")
     * @Groups ("post:read")
     * @Assert\Range(
     *      min = "now",
     *     max= "+30days",
     *      notInRangeMessage = "La date doit etre supperieur a notre date actuelle {{ min }} et ne doit pas depasser {{ max }} ",
     * )
     */
    private $dateEvent;

    /**
     * @var string|null
     *
     * @ORM\Column(name="descriptions", type="string", length=255, nullable=true)
     * @Groups ("post:read")
     */
    private $descriptions;



    /**
     * @var \CategorieEvent
     *
     * @ORM\ManyToOne(targetEntity=CategorieEvent::class, inversedBy="evenement")
     * @ORM\JoinColumns({
     * @ORM\JoinColumn(name="category", referencedColumnName="id_categ_event")
     * })
     *@Groups ("post:read")
     */
    private $category;

    /**
     * @return int
     */
    public function getIdEvent(): ?int
    {
        return $this->idEvent;
    }

    /**
     * @param int $idEvent
     */
    public function setIdEvent(int $idEvent): void
    {
        $this->idEvent = $idEvent;
    }

    /**
     * @return int|null
     */
    public function getIdCoach(): ?int
    {
        return $this->idCoach;
    }

    /**
     * @param int|null $idCoach
     */
    public function setIdCoach(?int $idCoach): void
    {
        $this->idCoach = $idCoach;
    }

    /**
     * @return string|null
     */
    public function getLieu(): ?string
    {
        return $this->lieu;
    }

    /**
     * @param string|null $lieu
     */
    public function setLieu(?string $lieu): void
    {
        $this->lieu = $lieu;
    }

    /**
     * @return int|null
     */
    public function getNbPlace(): ?int
    {
        return $this->nbPlace;
    }

    /**
     * @param int|null $nbPlace
     */
    public function setNbPlace(?int $nbPlace): void
    {
        $this->nbPlace = $nbPlace;
    }

    /**
     * @return \DateTime|null
     */
    public function getDateEvent(): ?\DateTime
    {
        return $this->dateEvent;

    }

    /**
     * @param \DateTime|null $dateEvent
     */
    public function setDateEvent(?\DateTime $dateEvent): void
    {
        $this->dateEvent = $dateEvent;
    }

    /**
     * @return string|null
     */
    public function getdescriptions(): ?string
    {
        return $this->descriptions;
    }

    /**
     * @param string|null $descriptions
     */
    public function setdescriptions(?string $descriptions): void
    {
        $this->descriptions = $descriptions;
    }

    /**
     * @return string|null
     */
    public function getDescription(): ?string
    {
        return $this->description;
    }

    /**
     * @param string|null $description
     */
    public function setDescription(?string $description): void
    {
        $this->description = $description;
    }

    /**
     * @return CategorieEvent
     */
    public function getcategory(): ?CategorieEvent
    {
        return $this->category;
    }

    /**
     * @param CategorieEvent $category
     */
    public function setcategory(CategorieEvent $category): void
    {
        $this->category = $category;
    }
    public function __toString() {
        return $this->description;
    }


}
